### DISCUSSION TITLE
Abstraction

### DISCUSSION THREAD

**Discussion by robstratton** [JoshClose/CsvHelper#2116] [Ideas]
Hi Folks, 

Loving your project and it's very powerful.  I was wondering if there are any plans to break the project up in a way that would allow use in a more separated solution.  Specifically, I was looking for an abstractions library that would give me the ability to use attributes, class maps, etc. without having to import the entire library - that would be done in a different project.  For now, I've imported the entire project, but would love to cut down on dependencies and retain my separation of concerns.

**Comment by JoshClose** [comment]
Would it still have the dependency of `ICsvParser` or are you thinking the map/attributes/etc would be something more generic?

**Comment by robstratton** [comment]
Hi Josh,

Based on my original needs, I was attempting to utilize the attributes and map definitions from a project that was abstracted from my parser library.   I have domain entities and DTO which I would have liked to decorate with attributes, but really did not want to import the entire CsvHelper library into that particular solution and chose a different path.

That said, I can see having ICsvParser also in such a library, so it could be injected into my various classes. Today, I would create my own interface that defined the methods  and properties I'd use in my project and wrap / extend the CsvHelper parser to do what I wanted.  

That's not quite a big deal, but I do like how easy something like SeriLog is to use as it implements the native ILogger interface.  Being able to do something similar here would be fabulous, but my current need is the attributes so I can remove the CsvHelper dependency from my library that doesn't (and won't) do any parsing.
